src/pages/index.js

```diff
@@ -1,96 +1,178 @@
+/*                       Create and Initialize new Classes                    */
 /*----------------------------------------------------------------------------*/
-// Section
-const cardsList = new Section(
-  {
-    data: initialCards,
-    renderer: renderCard,
+// API
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "0fe9b689-fa89-45da-98e3-26ffd27d7799",
+    "Content-Type": "application/json",
   },
-  cardElements
-);
+});
+
+// UserInfo
+const newUser = new UserInfo({
+  nameSelector: ".profile__title",
+  jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
 
 // Popups
 const popupCardForm = new PopupWithForm("#add-modal", handleAddFormSubmit);
 const popupNameForm = new PopupWithForm("#edit-modal", handleProfileFormSubmit);
 const popupWithImage = new PopupWithImage("#image-modal");
-// UserInfo
+const popupConfirmForm = new PopupWithConfirm(
+  "#confirm-modal",
+  handleConfirmFormSubmit
+);
+const popupChangeForm = new PopupWithForm(
+  "#change-modal",
+  handleChangeFormSubmit
+);
 
 /*----------------------------------------------------------------------------*/
-/*                        Initialize the instances                            */
+/*                        Initial Cards and userData                          */
 /*----------------------------------------------------------------------------*/
-// Initialize all my instances
-cardsList.renderItems();
-
-// All the rest
+let cardsList, myId;
+Promise.all([api.getUserInfo(), api.getInitialCards()]).then(
+  ([userData, cardsData]) => {
+    myId = userData._id;
+    newUser.setUserInfo(userData);
+    cardsList = new Section(
+      {
+        data: cardsData,
+        renderer: renderCard,
+      },
+      cardElements
+    );
+    cardsList.renderItems(cardsData);
+  }
+);
 
 /*----------------------------------------------------------------------------*/
 /*                                  Validation                                */
 /*----------------------------------------------------------------------------*/
 
 const editValidator = new FormValidator(validationSettings, profileForm);
 const addValidator = new FormValidator(validationSettings, cardForm);
+// const confirmValidator = new FormValidator(validationSettings, confirmForm);
+const changeValidator = new FormValidator(validationSettings, changeForm);
 editValidator.enableValidation();
 addValidator.enableValidation();
+changeValidator.enableValidation();
 
 /*----------------------------------------------------------------------------*/
 /*                                  Functions                                 */
 /*----------------------------------------------------------------------------*/
 
+async function handleLikeClick(card) {
+  const isLiked = card.cardIsLiked();
+  await api
+    .changeLikeStatus(card._cardId, isLiked)
+    .then((data) => {
+      this.setLikeCounter(data.likes);
```

Comment:
`this.`

`this` should **not** be used outside classes in functions in `index.js` . This context is easy to lose. You need to find other ways of passing data.

`card` from line 97 is the context `this` 

```js

      card.setLikeCounter(data.likes);

```